### DISCUSSION TITLE
KNX -> SP112 tasmota -> stops working after a while - IGMP issue?

### DISCUSSION THREAD

**Discussion by Zaschii** [arendst/Tasmota#20623] [Support request]
### PROBLEM DESCRIPTION
usually same issue then in this old topic https://github.com/arendst/Tasmota/issues/5425.
I have 2 Gosund SP112 plugs, flashed with tasmota 13.2 knx version.
Both plugs have the same problem.
On our MDT Glastaster / KNX switches we have buttons to switch this tasmota plugs on / off directly via KNX. 
Everything works for a while (few hours)... then the switch works only sporadically. It seems that packages are lost.
if i then unplug the socket and plug it in again, everything works again for a while.
or wlan is based on unifi ap which are connected per poe in a zyxel gs1900 poe switch.
in this switch and in unifi igmp v3 and igmp snooping is activated.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Gosund SP112
- [x] Tasmota binary firmware version number used: KNX
  - [X] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
CMD: Backlog Template; Module; GPIO 255
10:46:01.712 RSL: RESULT = {"NAME":"Gosund 112v3.4","GPIO":[320,0,321,0,2656,2720,0,0,2624,257,224,0,0,4800],"FLAG":0,"BASE":18}
10:46:01.924 RSL: RESULT = {"Module":{"0":"Gosund 112v3.4"}}
10:46:02.129 RSL: RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"0":"None"},"GPIO2":{"321":"Led_i2"},"GPIO3":{"0":"None"},"GPIO4":{"2656":"HLWBL CF1"},"GPIO5":{"2720":"BL0937 CF"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"2624":"HLWBL SEL_i"},"GPIO13":{"257":"Relay_i2"},"GPIO14":{"224":"Relay1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"4800":"ADC Button"}}


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
10:47:11.590 CMD: Backlog Rule1; Rule2; Rule3
10:47:11.680 RSL: RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
10:47:11.891 RSL: RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
10:47:12.140 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
10:47:40.221 CMD: Status 0
10:47:40.228 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"SP112-1","FriendlyName":["SP112-1","Tasmota2"],"Topic":"tasmota_02E033","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":0,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
10:47:40.261 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-knx.bin","RestartReason":"Software/System restart","Uptime":"0T00:02:00","StartupUTC":"2024-01-30T09:45:40","Sleep":50,"CfgHolder":4617,"BootCount":30,"BCResetTime":"2020-12-04T17:29:20","SaveCount":42,"SaveAddress":"F6000"}}
10:47:40.294 RSL: STATUS2 = {"StatusFWR":{"Version":"13.2.0(knx)","BuildDateTime":"2023-10-19T09:02:01","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"389/699"}}
10:47:40.313 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":2,"LogHost":"","LogPort":514,"SSId":["Willson-WLAN",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["0A008001","2805C80001000600003C5A0A002800000000","00000080","00006000","00004000","00000000"]}}
10:47:40.346 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":631,"Free":372,"Heap":21,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","9F9AC587","04268011","000000CF","010013C0","8000F181","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,11,12,16,18,19,22,24,26,27,30,35,37,45,62,68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
10:47:40.380 RSL: STATUS5 = {"StatusNET":{"Hostname":"SP112-1","IPAddress":"192.168.178.76","Gateway":"192.168.178.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.178.1","DNSServer2":"0.0.0.0","Mac":"B8:F0:09:02:E0:33","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
10:47:40.404 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-01-30T09:47:40","Local":"2024-01-30T10:47:40","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"08:23","Sunset":"17:42"}}
10:47:40.421 RSL: STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
10:47:40.439 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-01-30T10:47:40","ENERGY":{"TotalStartTime":"2020-12-04T17:29:20","Total":0.321,"Yesterday":0.013,"Today":0.008,"Power":0,"ApparentPower":0,"ReactivePower":0,"Factor":0.00,"Voltage":0,"Current":0.000}}}
10:47:40.458 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-01-30T10:47:40","Uptime":"0T00:02:00","UptimeSec":120,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Willson-WLAN","BSSId":"24:5A:4C:5F:20:B1","Channel":1,"Mode":"11n","RSSI":20,"Signal":-90,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
see description

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
![KNX_config_sp112](https://github.com/arendst/Tasmota/assets/54997850/4f8e4381-cddd-46ed-baaf-f2671a146fd0)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Sadly to say there a lot of devices which have a not reliable working IGMP implementation.
Have you checked if the packets are still transmitted?
Tasmota KNX can't work when packages are lost / not sent / dropped.
Sorry without more detailed info where can be seen that packages are arriving correctly we can't help.

**Comment by Zaschii** [comment]
igmp as cause is just a assumption from my side.
i can see that the group adress is correctly sent on the knx bus. 

I tried a multicast tester app and if i hit the knx button i see some cryptic values in this app - so iam not sure what happens with the packages.
How can i track the traffic effective? maybe with ntopng?

**Comment by Jason2866** [comment]
Maybe this helps https://gist.github.com/friek/6c10535b2e43f0e0867ce21b0679a337 to test.

**Comment by Jason2866** [comment]
To monitor wifi traffic with Wireshark is tricky.
